server/mailet/dkim/src/test/java/org/apache/james/jdkim/mailets/ConvertTo7BitTest.java

```diff
@@ -0,0 +1,242 @@
+
+                SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                   assertThat(mail.getMessage().getHeader(CONTENT_TRANSFER_ENCODING))
+                       .containsOnly(QUOTED_PRINTABLE);
+                   assertThat(mail.getMessage().getHeader(X_MIME_AUTOCONVERTED))
+                       .containsOnly("abc");
+                }));
+            }
+
+            @Disabled("Current implementation doesn't satisfy this test:\n" +
+                "Expected :A 8bit encoded body with =E2=82=ACuro symbol.\n" +
+                "Actual   :A 8bit encoded body with €uro symbol.")
+            @Test
+            void serviceShouldConvertContentToQuotedPrintable() throws Exception {
+                Mail mail = FakeMail.builder()
+                    .name("a-mail-with-8bit-encoding")
+                    .mimeMessage(mimeMessage8bit())
+                    .build();
+                testee.service(mail);
+
+                assertThat(mail.getMessage().getContent())
+                    .isEqualTo(MESSAGE_BODY_QUOTED_PRINTABLE);
+            }
+        }
+
+        @Nested
+        class WhenMultipart {
+
+            @Test
+            void serviceShouldAlertTextPartHeaders() throws Exception {
+                Mail mail = FakeMail.builder()
+                    .name("a-mail-with-8bit-encoding")
+                    .mimeMessage(MimeMessageUtil.mimeMessageFromStream(
+                        ClassLoader.getSystemResourceAsStream("eml/multipart-8bit.eml")))
+                    .build();
+
+                testee.service(mail);
+
+                BodyPart textPart = ((MimeMultipart) mail.getMessage().getContent())
+                    .getBodyPart(0);
+
+                SoftAssertions.assertSoftly(Throwing.consumer(softly -> {
+                   assertThat(textPart.getHeader(CONTENT_TRANSFER_ENCODING))
+                       .containsOnly(QUOTED_PRINTABLE);
+                   assertThat(textPart.getHeader(X_MIME_AUTOCONVERTED))
+                       .containsOnly("abc");
+                }));
+            }
+
+            @Disabled("Current implementation doesn't satisfy this test:\n" +
+                "Expected :A 8bit encoded body with =E2=82=ACuro symbol.\n" +
+                "Actual   :A 8bit encoded body with €uro symbol.")
+            @Test
+            void serviceShouldConvertTextPartContentToQuotedPrintable() throws Exception {
+                Mail mail = FakeMail.builder()
+                    .name("a-mail-with-8bit-encoding")
+                    .mimeMessage(MimeMessageUtil.mimeMessageFromStream(
+                        ClassLoader.getSystemResourceAsStream("eml/multipart-8bit.eml")))
+                    .build();
+
+                testee.service(mail);
+
+                BodyPart textPart = ((MimeMultipart) mail.getMessage().getContent())
+                    .getBodyPart(0);
+                assertThat(textPart.getContent())
+                    .isEqualTo(MESSAGE_BODY_QUOTED_PRINTABLE);
```

Comment:
I would write this test with :

```
assertThat(MimeMessageUtil.asString(mail.getMessage()))
    .contains(MESSAGE_BODY_QUOTED_PRINTABLE);
```

Rationals: I think the `writeTo` Message method might be triggering some changes...